### PR TITLE
Changed the docker and k8 config providers to return logs config as well

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Updated the utils method to extract config from docker labels and kubernetes annotations to return logs config.

### Motivation

Share a common logic to extract configs from docker labels and k8 annotations.

### CODE REVIEW

**Review by arbll — DISMISSED** [review]

**Review by hkaj — DISMISSED** [review]

File: pkg/autodiscovery/autoconfig.go [hunk 1]
```diff
@@ -254,8 +252,13 @@ func (ac *AutoConfig) schedule(configs []integration.Config) {
 	ac.scheduler.Schedule(configs)
 }
 
-// storeAndResolve store (in template cache) and resolves a given config into a slice of resolved configs
-func (ac *AutoConfig) storeAndResolve(config integration.Config) []integration.Config {
+// unschedule takes a slice of configs and unschedule them
+func (ac *AutoConfig) unschedule(configs []integration.Config) {
```
Comment (mfpierre) [inline pkg/autodiscovery/autoconfig.go]: If you add this can you also call it in `processRemovedConfigs`?

Comment (ajacquemot) [reply pkg/autodiscovery/autoconfig.go]: 👍

**Review by hkaj — CHANGES_REQUESTED** [review]

File: pkg/autodiscovery/autoconfig.go [hunk 2]
```diff
@@ -698,4 +706,11 @@ func (ac *AutoConfig) processDelService(svc listeners.Service) {
 	ac.store.removeConfigsForService(svc.GetEntity())
 	ac.processRemovedConfigs(configs)
 	ac.store.removeTagsHashForService(svc.GetEntity())
+	// FIXME: unschedule remove services as well
+	ac.unschedule([]integration.Config{
+		{
+			LogsConfig: integration.Data{},
+			Entity:     svc.GetEntity(),
```
[the remark this answers is not in the document]
Comment (ajacquemot) [reply pkg/autodiscovery/autoconfig.go]: We don't need the creation time here but we could add it for consistency.

**Review by hkaj — APPROVED** [review]
nice! 🍪

File: pkg/autodiscovery/autoconfig.go [hunk 3]
```diff
@@ -676,19 +677,26 @@ func (ac *AutoConfig) processNewService(svc listeners.Service) {
...
-
 		// ask the Collector to schedule the checks
 		ac.schedule([]integration.Config{resolvedConfig})
 	}
+	// FIXME: schedule new services as well
+	ac.schedule([]integration.Config{
+		{
+			LogsConfig:   integration.Data{},
+			Entity:       svc.GetEntity(),
+			CreationTime: svc.GetCreationTime(),
```
[the remark this answers is not in the document]
Comment (ajacquemot) [reply pkg/autodiscovery/autoconfig.go]: Created a ticket for this, thanks for the feedback

**Pull request merged**